sqlobject/src/test/java/org/jdbi/v3/sqlobject/MockClock.java

```diff
@@ -16,27 +16,32 @@
 import java.time.Clock;
 import java.time.Instant;
 import java.time.ZoneId;
+import java.time.ZonedDateTime;
 import java.time.temporal.TemporalUnit;
 
-class MockClock extends Clock {
-    private Instant now = Instant.now();
+public class MockClock extends Clock {
+    private ZonedDateTime now;
+
+    public MockClock(ZonedDateTime now) {
+        this.now = now;
+    }
 
     @Override
     public ZoneId getZone() {
-        return ZoneId.systemDefault();
+        return now.getZone();
     }
 
     @Override
     public Clock withZone(ZoneId zone) {
-        throw new UnsupportedOperationException();
+        return new MockClock(now.withZoneSameInstant(zone));
     }
 
     @Override
     public Instant instant() {
-        return now;
+        return now.toInstant();
     }
 
-    public Instant advance(long amountToAdd, TemporalUnit unit) {
-        return now = now.plus(amountToAdd, unit);
+    public void advance(long amountToAdd, TemporalUnit unit) {
```

Comment:
Why not have this still return `Instant`?

Reply:
It returning an Instant was a nice coincidence with its implementation rather than the logical outcome of telling a clock to advance. It's easy to put back but I felt it oppurtunistic instead of deliberate.

Reply:
With its current impl it would return a ZonedDateTime instead, or need to convert it to an Instant to stay the same. That being weird code to have in that location tells me the premise itself is weird. Clocks already have proper API to return the time, and that's what should be used instead of conveniently returning it here because we can.

Does that convince you or do I put it back?

Reply:
How is the premise weird here?  This is a simple accumulate operation like `+=`, except that we operate on a time rather than an int.  Or comparable to `AtomicLong.incrementAndGet`.  This isn't a coincidence, it was intentional, and it made using the class nicer.

I'm somewhat against returning a `ZonedDateTime` here.  I think that makes the API less tailored to our use case -- `jdbi` code should be avoiding timezone handling as much as possible, and we don't want to encourage its spread.  Additionally, as you point out, the `Clock` interface returns `Instant`s.  So it's weird for a `MockClock` to then return a `ZonedDateTime` instead when a normal Clock only does Instant.